File: backend/ERP.Backend/Domain/Interfaces/IUseCases/ICategoriaUseCases.cs
using Domain.DTOs;
using Domain.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Domain.Interfaces.IUseCases
{
    /// <summary>
    /// Interfaz de los casos de uso de categorías de producto.
    /// </summary>
    public interface ICategoriaUseCases
    {
        /// <summary>
        /// Obtiene todas las categorías de producto mapeadas a DTO.
        /// </summary>
        /// <returns>Lista de <see cref="CategoriaDTO"/> con todos los registros.</returns>
        Task<List<CategoriaDTO>> GetAllAsync();

        /// <summary>
        /// Obtiene una categoría por su identificador único mapeada a DTO.
        /// </summary>
        /// <param name="categoriaID">Identificador único de la categoría a buscar.</param>
        /// <returns>El <see cref="CategoriaDTO"/> encontrado, o <c>null</c> si no existe.</returns>
        Task<CategoriaDTO?> GetByIdAsync(int categoriaID);
    }
}

File: frontend/src/app/features/pedidos/domain/interfaces/usecases/ICategoriaUseCases.ts
import { Categoria } from '../../entities/categoria';

/**
 * Contrato de los casos de uso de categoría.
 * @remarks Implementado por {@link CategoriaUseCases}.
 */
export interface ICategoriaUseCases {
  /** Devuelve todas las categorías. */
  getAllAsync(): Promise<Categoria[]>;
  /**
   * Obtiene una categoría por su identificador.
   * @param id Identificador de la categoría.
   * @returns Entidad {@link Categoria} o `null` si no existe.
   */
  getByIdAsync(id: number): Promise<Categoria | null>;
  /** Devuelve sólo las categorías activas. */
  getActivosAsync(): Promise<Categoria[]>;
}